src/main/sensors/battery.c

```diff
@@ -179,73 +145,25 @@ const char * getBatteryStateString(void)
     return batteryStateStrings[batteryState];
 }
 
-void batteryInit(void)
+uint8_t batteryVoltagePercentage(void)
 {
-    batteryState = BATTERY_NOT_PRESENT;
-    batteryCellCount = 1;
-    batteryWarningVoltage = 0;
-    batteryCriticalVoltage = 0;
-
-    BiQuadNewLpf(VBATT_LPF_FREQ, &vbatFilterState, 50000);
-
-}
-
-#define ADCVREF 3300   // in mV
-int32_t currentSensorToCentiamps(uint16_t src)
-{
-    int32_t millivolts;
-
-    millivolts = ((uint32_t)src * ADCVREF) / 4096;
-    millivolts -= batteryConfig()->currentMeterOffset;
-
-    return (millivolts * 1000) / (int32_t)batteryConfig()->currentMeterScale; // current in 0.01A steps
-}
-
-void updateCurrentMeter(int32_t lastUpdateAt)
-{
-#ifndef ADC_CURRENT
-    UNUSED(lastUpdateAt);
-#else
-    static int32_t amperageRaw = 0;
-    static int64_t mAhdrawnRaw = 0;
-
-    amperageRaw -= amperageRaw / 8;
-    amperageRaw += (amperageLatestADC = adcGetChannel(ADC_CURRENT));
-    amperage = currentSensorToCentiamps(amperageRaw / 8);
-
-    mAhdrawnRaw += (MAX(0, amperage) * lastUpdateAt) / 1000;
-    mAhDrawn = mAhdrawnRaw / (3600 * 100);
-#endif
+    return constrain((((uint32_t)vbat - (batteryConfig()->vbatmincellvoltage * batteryCellCount)) * 100) / ((batteryConfig()->vbatmaxcellvoltage - batteryConfig()->vbatmincellvoltage) * batteryCellCount), 0, 100);
 }
 
-
-void updateVirtualCurrentMeter(int32_t lastUpdateAt, throttleStatus_e throttleStatus)
+uint8_t batteryCapacityRemainingPercentage(void)
 {
-    static int64_t mAhdrawnRaw = 0;
-    int32_t throttleOffset = (int32_t)rcCommand[THROTTLE] - 1000;
-    int32_t throttleFactor = 0;
-
+    uint16_t batteryCapacity = batteryConfig()->batteryCapacity;
 
-    amperage = (int32_t)batteryConfig()->currentMeterOffset;
-    if (ARMING_FLAG(ARMED)) {
-        if (throttleStatus == THROTTLE_LOW && feature(FEATURE_MOTOR_STOP))
-            throttleOffset = 0;
-        throttleFactor = throttleOffset + (throttleOffset * throttleOffset / 50);
-        amperage += throttleFactor * (int32_t)batteryConfig()->currentMeterScale  / 1000;
-    }
+    currentMeter_t *state = getCurrentMeter(batteryConfig()->currentMeterSource);
 
-    mAhdrawnRaw += (MAX(0, amperage) * lastUpdateAt) / 1000;
-    mAhDrawn = mAhdrawnRaw / (3600 * 100);
+    return constrain((batteryCapacity - constrain(state->mAhDrawn, 0, 0xFFFF)) * 100.0f / batteryCapacity , 0, 100);
```

Comment:
`mAhDrawn` is IMO property of battery, not current meter (even at the cost of increasing code coupling)